components/director/internal/model/eventing.go

```diff
@@ -1,29 +1,80 @@
 package model
 
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/pkg/errors"
+)
+
 type EventingConfiguration struct {
-	DefaultURL string
+	DefaultURL url.URL
 }
 
 type RuntimeEventingConfiguration struct {
 	EventingConfiguration
 }
 
-func NewRuntimeEventingConfiguration(url string) *RuntimeEventingConfiguration {
+func NewRuntimeEventingConfiguration(eventURL string) (*RuntimeEventingConfiguration, error) {
+	validUrl, err := validateURL(eventURL)
+	if err != nil {
+		return nil, err
+	}
+
 	return &RuntimeEventingConfiguration{
 		EventingConfiguration: EventingConfiguration{
-			DefaultURL: url,
+			DefaultURL: validUrl,
 		},
-	}
+	}, nil
 }
 
+const ApplicationEventingURLScheme = "https://%s/%s/v1/events"
+
 type ApplicationEventingConfiguration struct {
 	EventingConfiguration
 }
 
-func NewApplicationEventingConfiguration(url string) *ApplicationEventingConfiguration {
+func NewApplicationEventingConfiguration(runtimeEventURL url.URL, appName string) (*ApplicationEventingConfiguration, error) {
+	var appEventURL string
+
+	if runtimeEventURL.Host == "" {
+		appEventURL = ""
+	} else {
+		appEventURL = fmt.Sprintf(ApplicationEventingURLScheme, runtimeEventURL.Host, appName)
+	}
+
+	validUrl, err := validateURL(appEventURL)
```

Comment:
Can we parse the URL and then add the path? It would be simpler approach, right?